lib/stat_tracker.rb

```diff
@@ -493,4 +491,51 @@ def worst_defense
     team_id = all_teams_opponent_averages.key(all_teams_opponent_averages.values.max)
     get_team_name_from_id(team_id)
   end
+
+  def find_games_by_team_id(team_id, games = @games)
+    games.select do |game|
+      game.away_team_id == team_id || game.home_team_id == team_id
+    end
+
+  end
+
+  def group_games_by_season_and_team(season_id, team_id)
+    all_games_in_season = games_by_season[season_id]
+    all_games_a_team_played = find_games_by_team_id(team_id)
+    all_games_in_season & all_games_a_team_played
```

Comment:
this line returns an array of the elements that are common to those two arrays. 
So, I first grouped all of the games by season_id and put that in an array. Then the same with all the games a team played. Finally, I returned the elements that were common to both.